packages/contracts-bedrock/contracts/test/SystemConfig.t.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { CommonTest } from "./CommonTest.t.sol";
+import { SystemConfig } from "../L1/SystemConfig.sol";
+
+contract SystemConfig_Init is CommonTest {
+    SystemConfig sysConf;
+
+    function setUp() external {
+        sysConf = new SystemConfig({
+            _owner: alice,
+            _overhead: 2100,
+            _scalar: 1000000,
+            _batcherHash: bytes32(hex"abcd"),
+            _gasLimit: 9_000_000
+        });
+    }
+}
+
+contract SystemConfig_Initialize_TestFail is CommonTest {
+    function test_initialize_lowGasLimit_reverts() external {
+        vm.expectRevert("SystemConfig: gas limit too low");
+        new SystemConfig({
+            _owner: alice,
+            _overhead: 0,
+            _scalar: 0,
+            _batcherHash: bytes32(hex""),
+            _gasLimit: 7_999_999
+        });
+    }
+}
+
+contract SystemConfig_Setters_TestFail is SystemConfig_Init {
+    function test_setBatcherHash_notOwner_reverts() external {
+        vm.expectRevert("Ownable: caller is not the owner");
+        sysConf.setBatcherHash(bytes32(hex""));
+    }
+
+    function test_setGasConfig_notOwner_reverts() external {
+        vm.expectRevert("Ownable: caller is not the owner");
+        sysConf.setGasConfig(0, 0);
+    }
+
+    function test_setGasLimit_notOwner_reverts() external {
+        vm.expectRevert("Ownable: caller is not the owner");
+        sysConf.setGasLimit(0);
+    }
+}
+
+contract SystemConfig_Setters_Test is SystemConfig_Init {
+    event ConfigUpdate(
+        uint256 indexed version,
+        SystemConfig.UpdateType indexed updateType,
+        bytes data
+    );
+
+    function testFuzz_setBatcherHash_succeeds(bytes32 newBatcherHash) external {
+        vm.expectEmit(true, true, true, true);
+        emit ConfigUpdate(0, SystemConfig.UpdateType.BATCHER, abi.encode(newBatcherHash));
+
+        vm.prank(alice);
+        sysConf.setBatcherHash(newBatcherHash);
+        assertEq(sysConf.batcherHash(), newBatcherHash);
+    }
+
+    function testFuzz_setGasConfig_succeeds(uint256 newOverhead, uint256 newScalar) external {
+        vm.expectEmit(true, true, true, true);
+        emit ConfigUpdate(
+            0,
+            SystemConfig.UpdateType.GAS_CONFIG,
+            abi.encode(newOverhead, newScalar)
+        );
+
+        vm.prank(alice);
```

Comment:
nit: `vm.prank(sysConf.owner())`

Makes it less contextual when reading and decouples the deploy constructor args

Reply:
good call